repos/system_upgrade/common/actors/detectcorruptedgrubenv/actor.py

```diff
@@ -0,0 +1,43 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.libraries.common.config import architecture
+from leapp.libraries.actor.detectcorruptedgrubenv import is_grubenv_corrupted
+from leapp.models import GrubConfigError, FirmwareFacts
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class DetectCorruptedGrubenv(Actor):
+    """
+    Check the grubenv config file has a 1K size and don't end with a line feed
+    """
+
+    name = 'detect_corrupted_grubenv'
+    consumes = (FirmwareFacts,)
+    produces = (Report, GrubConfigError)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        if architecture.matches_architecture(architecture.ARCH_S390X):
+            return
+        config = '/boot/grub2/grubenv'
+        for fact in self.consume(FirmwareFacts):
+            if fact.firmware == 'efi':
+                config = '/boot/efi/EFI/redhat/grubenv'
+                break
+        if is_grubenv_corrupted(config):
+            create_report([
+                reporting.Title('Detected a corrupted grubenv file.'),
+                reporting.Summary(
+                    'The file %s is corrupted. '
+                    'Please delete it and then regenerate it using grub2-mkconfig.'
+                    % config
+                ),
```

Comment:
thanks for the contribution!!
we should provide a little bit better explanation here what the *corrupted* means (i do not mean to have a dynamic text in this case, but rather a generic explanation `(the file has to have exact size and cannot end with the newline)  The corruption could be caused e.g. by a manual modification of the file which is not recommended.` and adding the information that the file must be valid to pass the upgrade correctly. The rest of the text should be under `reporting.Remediation` (with the hint parameter), to explan people what they are expected to do (skipping the "Please" word), e.g. `Regenerate the file using the grub2-mkconfig tool.` In case it's expect the command to be always same and safe, the command could be mentioned covered by `commands` field.
```python
reporting.Remediation(hint="text", commands=[[cmd, 'param', ......]])
```

Writting in rush now, I/We will look at it properly during the next two weeks. I am not sure now whether this will work well for hybrid boots as the configuration files have different paths based not only on the thing whether the efi is or not detected.